client/app.zig

```diff
@@ -567,6 +573,21 @@ pub fn frameLoop(self: *App) !void {
                 }
             }
 
+            if (!self.game.load_state.full_init) {
+                const phase = &self.game.load_state.phase;
+                const loadstate_render = imgui_t.begin("Loading...", &imgui.loadstate_open, 0);
+                defer imgui_t.end();
+                if (loadstate_render) {
+                    // SAFETY: we null-terminate the phase names when deserializing from FE
+                    imgui_t.text(@ptrCast(phase.phase_name.constSlice()));
+                    if (phase.sub_phase_name.len != 0)
+                        imgui_t.text(@ptrCast(phase.sub_phase_name.constSlice()));
```

Comment:
This syntax asserts to the compiler that there's a `0` after the string, and it's safety-checked illegal behaviour for there *not* to be.

```suggestion
                    imgui_t.text(phase.phase_name.buffer[0..phase.phase_name.len:0]);
                    if (phase.sub_phase_name.len != 0)
                        imgui_t.text(phase.sub_phase.len.buffer[0..phase.sub_phase.len:0]);
```